### PR TITLE
Infra for postgres database

### PR DESCRIPTION
**Pull request by zekehuntergreen**
Co-Authored-By: Marjan Kalanaki <15894063+marjisound@users.noreply.github.com>
Co-Authored-By: Philip McMahon <philip.mcmahon@guardian.co.uk>

## What does this change?
To improve observability we'd like to keep track of ingestion progress with a postgres database.
This PR
- creates a CDK project in `postgres/cdk` initialized with the [GuCDK CLI tool](https://github.com/guardian/cdk/blob/main/docs/setting-up-a-gucdk-project.md#creating-a-new-project)
- adds CDK for postgres observability database and its security groups
- adds scripts in `scripts/db/` to connect to local, CODE and PROD databases
- adds github action to deploy any CDK changes 

## How to test
- run `./scripts/db/psql-code.sh` to connect to the CODE database

### CODE REVIEW

File: postgres/cdk/lib/giant.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+		databaseSecurityGroup.addIngressRule(
+			dbAccessSecurityGroup,
+			Port.tcp(DATABASE_PORT),
+		);
+
+		new CfnOutput(this, 'DbAccessSecurityGroup', {
+			exportName: `${props.stack}-postgres-${props.stage}-DbAccessSecurityGroup`,
+			value: dbAccessSecurityGroup.securityGroupId,
+		});
```
Comment (philmcmahon) [inline postgres/cdk/lib/giant.ts:101]: @marjisound we could also export the db secret ARN (or just the name) here so we don't need to hardcode it in the other stack

File: postgres/scripts/build.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
```
Comment (philmcmahon) [inline postgres/scripts/build.sh:1]: I think this file is redundant now

File: scripts/db/psql-local.sh [hunk 3]
```diff
@@ -0,0 +1,2 @@
+#/usr/bin/env bash
+PGPASSWORD=lurch psql -h localhost -p 9432 -U lurch_master -d lurch "$@"
```
Comment (philmcmahon) [inline scripts/db/psql-local.sh:2]: needs giantifying

**Comment by philmcmahon** [comment]
One or two things to address but otherwise looking good!

File: postgres/cdk/lib/giant.ts (hunk 1 above)
Comment (marjisound) [reply postgres/cdk/lib/giant.ts:101]: great suggestion. Just added the PostgresDbSecretArn output and updated the policy in the investigations cloud formation https://github.com/guardian/investigations-platform/pull/445/files#diff-b73b95cdf4b7835c57995ae97371a13231b3c2af8404d7086027e13161191ef0R366

**Review by philmcmahon — APPROVED** [review]
Hooray!

infrastructure

postgres elephant

**Pull request merged**